EUR-Lex.js

```diff
@@ -103,96 +135,125 @@ function doWeb(doc, url) {
 			for (var i in items) {
 				articles.push(i);
 			}
-			ZU.processDocuments(articles, scrape);
+			return ZU.processDocuments(articles, scrape);
 		});
-	} else {
-		scrape(doc, url);
 	}
+	else if (detectWeb(doc, url) === "case") {
+		return scrapeCase(doc, url);
+	}
+	return scrape(doc, url);
 }
 
 
 // this maps language codes from ISO 639-1 to 639-3
 var languageMapping = {
-	'BG': 'bul',
-	'CS': 'ces',
-	'DA': 'dan',
-	'DE': 'deu',
-	'EL': 'ell',
-	'EN': 'eng',
-	'ES': 'spa',
-	'ET': 'est',
-	'FI': 'fin',
-	'FR': 'fra',
-	'GA': 'gle',
-	'HR': 'hrv',
-	'HU': 'hun',
-	'IT': 'ita',
-	'LV': 'lav',
-	'LT': 'lit',
-	'MT': 'mlt',
-	'NL': 'nld',
-	'PL': 'pol',
-	'PT': 'por',
-	'RO': 'ron',
-	'SK': 'slk',
-	'SL': 'slv',
-	'SV': 'swe'
+	BG: 'bul',
+	CS: 'ces',
+	DA: 'dan',
+	DE: 'deu',
+	EL: 'ell',
+	EN: 'eng',
+	ES: 'spa',
+	ET: 'est',
+	FI: 'fin',
+	FR: 'fra',
+	GA: 'gle',
+	HR: 'hrv',
+	HU: 'hun',
+	IT: 'ita',
+	LV: 'lav',
+	LT: 'lit',
+	MT: 'mlt',
+	NL: 'nld',
+	PL: 'pol',
+	PT: 'por',
+	RO: 'ron',
+	SK: 'slk',
+	SL: 'slv',
+	SV: 'swe'
 };
 
 
 function scrape(doc, url) {
 	var type = detectWeb(doc, url);
 	var item = new Zotero.Item(type);
-	
+
 	// determine the language we are currently looking the document at
-	var languageUrl = url.split('/')[4];
-	if (languageUrl=="AUTO") {
+	let languageUrl = url.split('/')[4];
+	if (languageUrl === "AUTO") {
 		languageUrl = autoLanguage || "EN";
 	}
-	var language = languageMapping[languageUrl] || "eng";
-	
+	const language = languageMapping[languageUrl] || "eng";
```

Comment:
Why const?

Reply:
It's shouldn't be mutated at any point. This makes it obvious if it is inadvertently mutated.